flax/nnx/nn/linear.py

```diff
@@ -163,6 +167,8 @@ def __init__(
     dot_general_cls: tp.Any = None,
     preferred_element_type: Dtype | None = None,
     rngs: rnglib.Rngs,
+    kernel_metadata: dict[str, tp.Any] | None = None,
```

Comment:
Can we make this type `tp.Mapping[str, tp.Any]` and default to an empty immutable mapping e.g. MappingProxyType? This way we can remove the None handling logic.